src/Microsoft.Data.Analysis/PrimitiveColumnContainer.cs

```diff
@@ -190,64 +198,69 @@ public void AppendMany(T? value, long count)
                 NullCount += count;
             }
 
-            while (count > 0)
+            var remaining = count;
+            while (remaining > 0)
             {
                 if (Buffers.Count == 0)
                 {
                     Buffers.Add(new DataFrameBuffer<T>());
                     NullBitMapBuffers.Add(new DataFrameBuffer<byte>());
                 }
-                int bufferIndex = Buffers.Count - 1;
-                ReadOnlyDataFrameBuffer<T> lastBuffer = Buffers[bufferIndex];
-                if (lastBuffer.Length == ReadOnlyDataFrameBuffer<T>.MaxCapacity)
+
+                if (Buffers[Buffers.Count - 1].Length == ReadOnlyDataFrameBuffer<T>.MaxCapacity)
                 {
-                    lastBuffer = new DataFrameBuffer<T>();
-                    Buffers.Add(lastBuffer);
+                    Buffers.Add(new DataFrameBuffer<T>());
                     NullBitMapBuffers.Add(new DataFrameBuffer<byte>());
                 }
-                DataFrameBuffer<T> mutableLastBuffer = DataFrameBuffer<T>.GetMutableBuffer(lastBuffer);
-                Buffers[bufferIndex] = mutableLastBuffer;
-                int allocatable = (int)Math.Min(count, ReadOnlyDataFrameBuffer<T>.MaxCapacity);
+
+
+                DataFrameBuffer<T> mutableLastBuffer = Buffers.GetMutable(Buffers.Count - 1);
+
+                int allocatable = (int)Math.Min(remaining, ReadOnlyDataFrameBuffer<T>.MaxCapacity);
+
                 mutableLastBuffer.EnsureCapacity(allocatable);
-                mutableLastBuffer.RawSpan.Slice(lastBuffer.Length, allocatable).Fill(value ?? default);
-                mutableLastBuffer.Length += allocatable;
-                Length += allocatable;
+                mutableLastBuffer.Length = allocatable;
+
+                DataFrameBuffer<byte> lastNullBitMapBuffer = NullBitMapBuffers.GetMutable(NullBitMapBuffers.Count - 1);
+                int nullBufferAllocatable = (allocatable + 7) / 8;
+                lastNullBitMapBuffer.EnsureCapacity(nullBufferAllocatable);
+                lastNullBitMapBuffer.Length = nullBufferAllocatable;
 
-                int nullBitMapBufferIndex = NullBitMapBuffers.Count - 1;
-                ReadOnlyDataFrameBuffer<byte> lastNullBitMapBuffer = NullBitMapBuffers[nullBitMapBufferIndex];
-                DataFrameBuffer<byte> mutableLastNullBitMapBuffer = DataFrameBuffer<byte>.GetMutableBuffer(lastNullBitMapBuffer);
-                NullBitMapBuffers[nullBitMapBufferIndex] = mutableLastNullBitMapBuffer;
-                int nullBitMapAllocatable = (int)(((uint)allocatable) / 8) + 1;
-                mutableLastNullBitMapBuffer.EnsureCapacity(nullBitMapAllocatable);
-                _modifyNullCountWhileIndexing = false;
-                for (long i = Length - count; i < Length; i++)
+                remaining -= allocatable;
+                Length += mutableLastBuffer.Length;
+
+                // PR Question: Does this need to be called if it's value is null/doesn't have value?
```

Comment:
This made an assumption that this code below is pointless to call `Fill` with  `default`.

```cs
mutableLastBuffer.RawSpan.Slice(mutableLastBuffer.Length, allocatable).Fill(value ?? default);
```

This also made an assumption that calling `SetValidityBit` is pointless when `value` is null.